### PR TITLE
Implement Volunteer Enrollment

### PR DESCRIPTION
**Pull request by brandonwong368**
### Notion ticket link

[Volunteer Enrollment](https://www.notion.so/uwblueprintexecs/086c1b564dbf478faa649a8880e34b81?v=9d3f8002a0d1477ba395616dfc082af7&p=50e0e899d4d94ba7a0fc526c688b749a)

### Implementation description

- separated media release, background check forms instead separate components
- confirmation page component, retrieves selected class to generate confirmation card. changed getClass get api to include program info
- for submit/update background check, redirects to document upload page, after successful upload, closing the document upload page will redirect to the next page in the flow
- upon finishing enrollment, a record is added to the VolunteerRegs table

https://user-images.githubusercontent.com/71860803/137991232-d4090311-0acd-4737-97bf-baba888992eb.mp4

### Steps to test

1. Log in, ensure that your user has the volunteer role.
2. Register through the /program-details/[pid] route using one of the program cards
3. Depending on the values of criminalCheckLink and criminalCheckExpired, either a Submit/Update background check will apear
4. When upload criminal record is clicked, should direct to document upload form. After a file is submitted, the X (Close button) should go back to the enrollment flow, skipping to the next page (Media Release)
5. Click through the separate pages
6. After clicking Finish, a new db record in VolunteerRegs should be created

### What should reviewers focus on?

- 

### Checklist

-   [x] My PR name is descriptive and in imperative tense
-   [x] I have run the linter
-   [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: src/components/volunteer-enroll/UpdateCriminalCheck.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/UpdateCriminalCheck.tsx]: For href, we always want to specify absolute path otherwise i18n subpaths won't be happy. We can also make the page redirect a bit more robust
```suggestion
                        href={`/document-upload?type=criminal-check&redirect=/volunteer/enrollment?classId=${classInfo.id}%26page=1`}
```

File: pages/volunteer/enrollment.tsx
Comment (MatoPlus) [inline pages/volunteer/enrollment.tsx]: The only item that differs here is the first item, We can apply the ternary on the first item only!

File: src/components/volunteer-enroll/ConfirmClass.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/ConfirmClass.tsx:35]: let's add some `marginBottom` on this button, it's hugging the footer too tight!

**Review by bekahma — APPROVED** [review]
looks good :D

File: pages/volunteer/enrollment.tsx
Comment (MatoPlus) [inline pages/volunteer/enrollment.tsx]:
```suggestion
```
Don't think this is needed?

File: pages/volunteer/enrollment.tsx
Comment (MatoPlus) [inline pages/volunteer/enrollment.tsx]: hmm, I think you can make this page more like the others like this:
```suggestion
            onNext={() => {submitVolunteer(); nextPage();})
```

File: src/components/volunteer-enroll/ConfirmClass.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/ConfirmClass.tsx]: idem

File: src/components/volunteer-enroll/ConfirmClass.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/ConfirmClass.tsx]: idem

File: src/components/volunteer-enroll/SubmitCriminalCheck.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/SubmitCriminalCheck.tsx:50]: Please replace this with provided translation, by using `const {t} = useTranslation("form")` - note that you will need to include this json in `getServerSideProps` in the base page - `...(await serverSideTranslations(context.locale, ["common", "form"])),`
```suggestion
                        t("bgc.missing")
```

File: src/components/volunteer-enroll/SubmitCriminalCheck.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/SubmitCriminalCheck.tsx:58]: idem, translate - `desc1`

File: src/components/volunteer-enroll/SubmitCriminalCheck.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/SubmitCriminalCheck.tsx:66]: idem, use translate - `desc2`

File: src/components/volunteer-enroll/VolunteerEnrollFormWrapper.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/VolunteerEnrollFormWrapper.tsx]: I was thinking this could be redirected to "/class"
```suggestion
                                    href="/class"
```

File: src/components/volunteer-enroll/VolunteerEnrollFormWrapper.tsx
Comment (MatoPlus) [inline src/components/volunteer-enroll/VolunteerEnrollFormWrapper.tsx]: This hover colour don't look right lol
image
```suggestion
```

**Review by MatoPlus — APPROVED** [review]
Yo you actually clapped this ticket xD. Great job! I'd be great if you can make my life easier and also add the translations for the following pages you implemented (This is optional btw, you can leave it to me after merging if you want, just thought it might be cool for you to dabble in):

All the translations you need is probably in the `form.json` locale. Specifically these properties:
```json
    "bgc": {
        "title": "Background Check",
        "submitTitle": "Submit a background check",
        "missing": "You have not uploaded a Background check yet!",
        "desc1": "As volunteering in our programs involves working closely with children and vulnerable persons, we ask that all our volunteers get a Criminal Record Check completed at their local RCMP office. The price of the CRC will be waived with a letter provided from SDC.",
        "desc2": "Also, please note that the MPM/JELIC is an IN-PERSON math program. If you apply to volunteer for this program, please ensure that you are aware that it is an in-person program and are able to attend the classes at Richmond Quantum Academy (6650-8181 Cambie Rd, Richmond, BC V6X 3X9).",
        "instruction": "Uploading your Background Check",
        "instruction1": "Under My Account > Background Check - Generate a volunteer letter from SDC",
        "instruction2": "Use the volunteer letter to obtain a criminal background check letter",
        "instruction3": "Upload a copy of the result to your SDC account.",
        "instruction4": "Once you’ve submitted your letter, keep an eye out for approval status from SDC!",
        "skip": "I don’t have a background check yet, but I will provide a CRC by the first day of volunteering.",
        "upload": "Upload background check",
        "updateTitle": "Update Background Check",
        "expired": "Your background check has expired!",
        "firstDayProvide": "I will provide a CRC by the first day of volunteering"
    },
    "form": {
        "accountCreated": "Account created successfully",
        "registered": "Thank you for registering!",
        "volunteerSignup": "Thank you for signing up to volunteer!",
        "browseClasses": "Browse Classes",
        "viewUpcoming": "View upcoming classes",
        "skip": "Skip for now",
        "next": "Next",
        "back": "Back"
    }
```


image

image

image

**Comment by MatoPlus** [comment]
This PR does not currently consider whether or not the volunteer has already registered or if the class is full. I will create another ticket to deal with this:

https://www.notion.so/uwblueprintexecs/Volunteer-Parent-Register-Class-Limit-No-multiple-registers-69ae85d521d94f7ebcaa665bff973855

This ticket is looking good to merge though.

**Pull request merged**
